src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1420,16 +1430,75 @@ private IEnumerable<TypeInferenceInfo> InferTypeInImplicitArrayCreation(Implicit
                 return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
             }
 
+            private IEnumerable<TypeInferenceInfo> InferTypeInPropertyPattern(PropertyPatternSyntax propertyPattern)
+            {
+                // something like: { ... sub patterns ... } or like: Blah { ... sub patterns ... }
+                // in the latter case, we know the type, because it's explicitly stated.  In the former
+                // we have to walk higher to figure it out.
+                var type = this.SemanticModel.GetTypeInfo(propertyPattern).ConvertedType;
+                return type == null
+                    ? SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>()
+                    : SpecializedCollections.SingletonEnumerable(new TypeInferenceInfo(type));
+            }
+
+            private IEnumerable<TypeInferenceInfo> InferTypeInConstantPattern(
+                ConstantPatternSyntax constantPattern)
+            {
+                return InferTypes(constantPattern);
+            }
+
+            private IEnumerable<TypeInferenceInfo> InferTypeInPropertySubpattern(
+                PropertySubpatternSyntax propertySubpattern,
+                SyntaxNode child)
+            {
+                return InferTypes(propertySubpattern);
+            }
+
+            private IEnumerable<TypeInferenceInfo> InferTypeInSubpatternElement(
+                SubpatternElementSyntax subpatternElement,
+                SyntaxNode child)
+            {
+                // we have  { X: ... }.  The type of ... is whatever the type of 'X' is in its
+                // parent type.  So look up the parent type first, then find the X member in it
+                // and use that type.
+                if (child == subpatternElement.Pattern &&
+                    subpatternElement.NameColon.Name is IdentifierNameSyntax identifier)
+                {
+                    var result = ArrayBuilder<TypeInferenceInfo>.GetInstance();
+
+                    foreach (var symbol in this.SemanticModel.GetSymbolInfo(subpatternElement.NameColon.Name).GetAllSymbols())
```

Comment:
nit: you just stored `subpatternElement.NameColon.Name` inside `identifier` above

Reply:
sure!